### PR TITLE
Enable Custom Grouping for DataIncrementalScenario

### PR DESCRIPTION
**Pull request by wistuba**
Rename `class_groupings` to `groupings` in `ClassIncrementalScenario`.

`DataIncrementalScenario` has now an argument `groupings` that allows grouping datasets similar to the `groupings` in `ClassIncrementalScenario` for classes.

Added support for training on `yearbook` with ViT.

Functionality exposed for benchmarking.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/renate/benchmark/experiment_config.py [hunk 1]
```diff
@@ -337,21 +355,38 @@ def train_transform(dataset_name: str) -> Optional[Callable]:
     if dataset_name in ["CLEAR10", "CLEAR100", "DomainNet"]:
         return transforms.Compose(
             [
-                transforms.ToTensor(),
-                transforms.Resize(
-                    224, interpolation=transforms.InterpolationMode.BILINEAR, antialias=True
-                ),
+                transforms.Resize(224),
```
[the remark this answers is not in the document]
Comment (wistuba) [reply src/renate/benchmark/experiment_config.py:358]: the input to resize is no longer a tensor but a pilimage. so this should be default: https://pytorch.org/vision/main/generated/torchvision.transforms.Resize.html

File: benchmarks/run_benchmark.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ def load_config(scenario_file, model_file, updater_file, dataset_file):
             instance_type="ml.g4dn.xlarge",
             n_workers=1,
             max_time=args.max_time,
+            instance_max_time=args.max_time,
```
Comment (prabhuteja12) [inline benchmarks/run_benchmark.py:115]: difference to `max_time`?

Comment (wistuba) [reply benchmarks/run_benchmark.py:115]: max_time refers to the tuning time per update step. instance_max_time refers to after which time the sagemaker training job is aborted.

File: test/renate/benchmark/test_scenarios.py [hunk 3]
```diff
@@ -49,42 +49,107 @@ def test_failing_to_init(tmpdir, scenario_cls, kwargs):
...
+def test_data_incremental_scenario_grouping():
+    """Test whether grouping in DataIncrementalScenario works"""
+    scenario = DataIncrementalScenario(
+        data_module=DummyDataIncrementalDataModule(0, (10, 1), val_size=0.2),
+        chunk_id=0,
+        groupings=((0, 1), (2,)),
+    )
+    scenario.prepare_data()
+    scenario.setup()
+    assert set(int(scenario.train_data()[i][1]) for i in range(80)) == {0}
```
Comment (prabhuteja12) [inline test/renate/benchmark/test_scenarios.py:91]: Can you write a comment about what is being tested for, here? I don't follow the 80, and following 20 are.

**Review by prabhuteja12 — APPROVED** [review]

**Pull request merged**